src/m365/spo/commands/tenant/tenant-applicationcustomizer-set.ts

```diff
@@ -0,0 +1,169 @@
+        newTitle: typeof args.options.newTitle !== 'undefined',
+        clientSideComponentProperties: typeof args.options.clientSideComponentProperties !== 'undefined',
+        webTemplate: typeof args.options.webTemplate !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --id [id]'
+      },
+      {
+        option: '-t, --title [title]'
+      },
+      {
+        option: '-c, --clientSideComponentId  [clientSideComponentId]'
+      },
+      {
+        option: '--newTitle [newTitle]'
+      },
+      {
+        option: '-p, --clientSideComponentProperties [clientSideComponentProperties]'
+      },
+      {
+        option: '-w, --webTemplate [webTemplate]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.id && isNaN(parseInt(args.options.id))) {
+          return `${args.options.id} is not a number`;
+        }
+
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        if (!args.options.newTitle && !args.options.clientSideComponentProperties && !args.options.webTemplate) {
+          return `Please specify an option to be updated`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['title', 'id', 'clientSideComponentId'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const appCatalogUrl = await spo.getTenantAppCatalogUrl(logger, this.debug);
+      if (!appCatalogUrl) {
+        throw 'No app catalog URL found';
+      }
+
+      const listItem = await this.getListItem(logger, args.options, appCatalogUrl);
+      await this.updateTenantWideExtension(appCatalogUrl, args.options, logger, listItem);
+    }
+    catch (err: any) {
+      return this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getListItem(logger: Logger, options: Options, appCatalogUrl: string): Promise<ListItemInstance> {
+    const { title, id, clientSideComponentId } = options;
+    const filter = title ? `Title eq '${title}'` : id ? `Id eq '${id}'` : `TenantWideExtensionComponentId eq '${clientSideComponentId}'`;
+
+    if (this.verbose) {
+      logger.logToStderr(`Getting tenant-wide application customizer: "${title || id || clientSideComponentId}"...`);
+    }
+
+    const listServerRelativeUrl: string = urlUtil.getServerRelativePath(appCatalogUrl, '/lists/TenantWideExtensions');
+    const listItemInstances = await odata.getAllItems<ListItemInstance>(`${appCatalogUrl}/_api/web/GetList('${formatting.encodeQueryParameter(listServerRelativeUrl)}')/items?$filter=TenantWideExtensionLocation eq 'ClientSideExtension.ApplicationCustomizer' and ${filter}`);
+    if (listItemInstances) {
```

Comment:
If you invert this if-statement and throw immediately on no items found, we can have less indentation on the other if-statements. Which would be more readable.